llm/data_gemma/rag.py

```diff
@@ -0,0 +1,130 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""RAG Flow."""
+
+import logging
+import time
+
+from data_gemma import base
+from data_gemma import dc
+from data_gemma import prompts
+from data_gemma import validate
+
+_MAX_QUESTIONS = 25
+
+
+class RAGFlow(base.Flow):
+  """Retrieval Augmented Generation."""
+
+  def __init__(
+      self,
+      llm_ques: base.LLM,
+      llm_ans: base.LLM,
+      datacommons: dc.DataCommons,
+      verbose: bool = True,
+      in_context: bool = False,
+      validate_dc_responses: bool = False,
+      metrics_list: str = '',
+  ):
+    self.llm_ques = llm_ques
+    self.llm_ans = llm_ans
+    self.dc = datacommons
```

Comment:
ver nit:

why not use datacommons.py, so here could be like

from data_gemma import datacommons
...
dc: datacommons.DataCommons
...
self.dc = dc

So the object name is always `dc`